simpleviewinc/tap-events-force
# Zen 629 price in select

**Ticket**: https://jira.simpleviewtools.com/browse/ZEN-629

## Context

To change the behavior of the price functionality 

## Goal

* Ensure the app matches templates as provided in https://www.figma.com/file/aLesACdNG3CvInUVuFm4hO/Session-selection---final
* Ensure price is available in the select button

## Updates

* Updated BookingButton.js to ensure text and price is displayed according to the rules :
 a. If the session has a price and is a SELECT state, the text displayed should change to BUY.
 b. If the session does not have a price and is a SELECT state, the text displayed should remain SELECT.
 c. If there is no cost , the button should not display any price or alt text like FREE
 d. If a session has a price but is ANY other state (e.g WAITING_LIST) no price should be displayed


## Testing

* Setup
  * Run the command => keg evf pack run evf:zen-629-priceinselect
  * Navigate to http://evf-ZEN-629-PriceInSelect.local.keghub.io
  Note : The current default test data should be enough to validate this ticket

* Testing all requirements
  * Ensure that for a select session with a price , the text on the button appears as 'BUY <price>'. 
  e.g : BUY $923
  * Ensure that for a select session without a price , the text on the button appears as just 'SELECT'
  * Ensure no FREE label appears anywhere for any state
  * Ensure for a waiting list , even though it has a price , no price is displayed . i.e. text should only be 'WAITING LIST' regardless of whether the session has a price or not
